sdk/src/CrashReporter.ts

```diff
@@ -127,33 +145,45 @@ export default class CrashReporter {
    * Cache a given Report to be sent later.
    * @param report - the Report to cache
    */
-  async saveCrashReport(report: CrashReportPayload): Promise<null> {
-    return RaygunNativeBridge.saveCrashReport(JSON.stringify(report));
+  async cacheCrashReport(report: CrashReportPayload): Promise<null> {
+    return RaygunNativeBridge.cacheCrashReport(JSON.stringify(report));
   }
 
   /**
-   * Load and return cached reports.
+   * Transmit cached reports.
+   * @param apiKey - The Raygun application to transmit too
+   * @param customEndpoint
    */
-  async loadCachedReports(): Promise<CrashReportPayload[]> {
-    return RaygunNativeBridge.loadCrashReports().then((reportsJson: string) => {
-      try {
-        return JSON.parse(reportsJson).filter(Boolean);
-      } catch (err) {
-        error(err);
-        return [];
-      }
-    });
+  async resendCachedReports(apiKey: string, customEndpoint?: string) {
+
+    //If there are Reports cached
+    if(!await RaygunNativeBridge.cacheEmpty()) {
+      //Extract cached reports from the native side
+      const cache : CrashReportPayload[] = await RaygunNativeBridge.flushCrashReportCache()
+      .then((reportsJson: string) => {
+        try {
+          //Format the cache and remove empty and null strings
+          return JSON.parse(reportsJson).filter(Boolean) as CrashReportPayload[];
+        } catch (err) {
+          //If there are no cached reports then return an empty array
+          error(err);
+          return [];
+        }
+      });
+
+      log('Cache flushed', cache);
+
+      //Attempt to send each of the cached reports
+      await Promise.all(cache.map(cachedReport => this.sendCrashReport(cachedReport, apiKey, customEndpoint)));
+    }
   }
 
   /**
-   * Transmit cached reports.
-   * @param apiKey - The Raygun application to transmit too
-   * @param customEndpoint
+   * Change the size of the local cache
```

Comment:
A more descriptive comment might be _"Sets the maximum number of Crash Reports that can be stored on the device."_